pm: device_runtime: Fix return values

ceolin opened the pull request:
Return -EBUSY in cases where the request failed because the reference
count does not reach the right value.

Fixes #37821

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
pm: device_runtime: Fix return values

Return 0 in cases where the request does not trigger a device state
change, only incremented or to decremented the reference count.

Fixes #37821

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

JordanYates reviewed (changes requested):
I elaborate in the linked issue, but the device still being in active mode because another user has requested it is not an error condition, and should not return an error code.

ceolin commented:
> I elaborate in the linked issue, but the device still being in active mode because another user has requested it is not an error condition, and should not return an error code.

That makes sense, initially I thought would be necessary to inform the caller the device didn't change to the requested state. EBUSY would serve this purpose, but this is probably more confuse and not necessary. Will change to just return success in these cases.

Commit:
pm: device_runtime: Fix possible underflow

The runtime API is referenced count and uses a uint32_t.
Avoid underflow when dealing with put requests.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>
